packages/gatsby-link/src/index.js

```diff
@@ -161,7 +163,10 @@ class GatsbyLink extends React.Component {
 
 GatsbyLink.propTypes = {
   ...NavLinkPropTypes,
-  innerRef: PropTypes.func,
+  innerRef: PropTypes.oneOfType([
```

Comment:
we could remove this proptype as it shouldn't be mandatory.

Reply:
Thanks for having a look, removed it! 🙂